src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,458 @@
+                          const std::string& imageFilePrefix,
+                          const std::string& imageFileExtension) {
+  textureTypeSanityCheck(flags_, type, "CubeMap::loadTexture():");
+
+  // set the alias of the texture
+  Mn::GL::CubeMapTexture* texture = nullptr;
+  switch (type) {
+    case TextureType::Color:
+      texture = textures_[TextureType::Color].get();
+      break;
+
+    case TextureType::Depth:
+      texture = textures_[TextureType::Depth].get();
+      break;
+  }
+  CORRADE_ASSERT(texture, "CubeMap::loadTexture(): Unknown texture type.", );
+
+  // plugin manager used to instantiate importers which in turn are used
+  // to load image data
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> manager;
+  std::string importerName{"AnyImageImporter"};
+  if (type == TextureType::Depth) {
+    importerName = "StbImageImporter";
+  }
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      manager.loadAndInstantiate(importerName);
+  CORRADE_INTERNAL_ASSERT(importer);
+  if (type == TextureType::Depth) {
+    // Override image channel count. Because depth info should be R32F
+    importer->configuration().setValue("forceChannelCount", 1);
+  }
+
+  const char* coordStrings[6] = {"+X", "-X", "+Y", "-Y", "+Z", "-Z"};
+  int imageSize = 0;
+
+  for (int iFace = 0; iFace < 6; ++iFace) {
+    // open image file
+
+    std::string filename = Cr::Utility::formatString(
+        "{}.{}.{}.{}", imageFilePrefix, getTextureTypeFilenameString(type),
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> imageData =
+        importer->image2D(0);
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(imageData);
+    Mn::Vector2i size = imageData->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    switch (type) {
+      case TextureType::Color:
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             *imageData);
+        break;
+
+      case TextureType::Depth: {
+        CORRADE_INTERNAL_ASSERT(imageData->format() == Mn::PixelFormat::R32F);
+        Mn::ImageView2D imageView(Mn::GL::PixelFormat::DepthComponent,
+                                  Mn::GL::PixelType::Float, imageData->size(),
+                                  imageData->data());
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             imageView);
+        // XXX
+        Mn::Debug{} << Cr::Containers::arrayCast<const float>(
+            imageView.data().prefix(32));
```

Comment:
Here, I output the first 12 floats from the loaded image. 
example outputs:
`{0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094, 0.996094}`
Everything is 0.996094 again.